### PR TITLE
done

### PR DESCRIPTION
**Pull request by NikitaKarbazin**

### CODE REVIEW

**Review by Kaunin1 — APPROVED** [review]
Interesting way of solution through the object.
Well done!

File: src/formatDate.js [hunk 1]
```diff
@@ -50,7 +50,45 @@
...
+        formatDateObject[symbol] = initialDateObject.hasOwnProperty(symbol)
+          ? initialDateObject[symbol]
+          : initialDateObject.YYYY.slice(2);
+        break;
+      case 'YYYY':
+        formatDateObject[symbol] = initialDateObject.hasOwnProperty(symbol)
+          ? initialDateObject[symbol]
+          : formatDateObject[symbol] = initialDateObject.YY >= 30
+            ? `19${initialDateObject.YY}`
+            : `20${initialDateObject.YY}`;
```
Comment (jstmpelowycz) [inline src/formatDate.js:84]: ternary overdose)

Comment (jstmpelowycz) [reply src/formatDate.js:84]:
```
if (initialDateObject.hasOwnProperty(symbol)) {
  formatDateObject[symbol] = initialDateObject[symbol];
} else {
 const prefix = initialDateObject.YY >= 30 ? 19 : 20;

	formatDateObject[symbol] = prefix + initialDateObject.YY;
}
```

Comment (jstmpelowycz) [reply src/formatDate.js:84]: I'd make it this way

File: src/formatDate.js [hunk 2]
```diff
@@ -50,7 +50,45 @@
...
+
+  const [separator, ...format] = fromFormat;
+  const initialDate = date
+    .split(separator)
+    .reverse()
+    .map((pieceOfDate, index) => {
+      return [format[index], pieceOfDate];
+    });
+
+  const initialDateObject = Object.fromEntries(initialDate);
```
Comment (jstmpelowycz) [inline src/formatDate.js:63]: 🔥

**Review by jstmpelowycz — APPROVED** [review]

**Pull request closed without merging**